### PR TITLE
Feature add optional algorithm query params

### PR DESCRIPTION
**Pull request by LHommeDeBat**
#### Short Description
Fronend Algorithm List view requires new QueryParameters.

#### Proposed Changes
- Added Sorting to Utils and implemented them within the "/algorithm" endpoint
- Added Search to "/algorithm" endpoint, allowing search of certain algorithm attributes

### CODE REVIEW

File: org.planqk.atlas.core/src/main/java/org/planqk/atlas/core/services/AlgorithmServiceImpl.java [hunk 1]
```diff
@@ -155,7 +156,10 @@ public void delete(UUID id) {
     }
 
     @Override
-    public Page<Algorithm> findAll(Pageable pageable) {
+    public Page<Algorithm> findAll(Pageable pageable, String search) {
+        if (!Objects.isNull(search) && !search.isEmpty()) {
+            return algorithmRepository.findByNameContainingIgnoreCaseOrAcronymContainingIgnoreCaseOrProblemContainingIgnoreCase(search, search, search, pageable);
```
Comment (c-mueller) [inline org.planqk.atlas.core/src/main/java/org/planqk/atlas/core/services/AlgorithmServiceImpl.java:161]: Does this potentially return duplicates?

Comment (LHommeDeBat) [reply org.planqk.atlas.core/src/main/java/org/planqk/atlas/core/services/AlgorithmServiceImpl.java:161]: Did not with test data. It just goes through each entry in DB and checks if it meets criteria.

File: org.planqk.atlas.core/src/main/java/org/planqk/atlas/core/repository/AlgorithmRepository.java [hunk 2]
```diff
@@ -43,6 +45,8 @@
 
     boolean existsAlgorithmById(UUID id);
 
+    Page<Algorithm> findByNameContainingIgnoreCaseOrAcronymContainingIgnoreCaseOrProblemContainingIgnoreCase(String name, String acronym, String problem, Pageable pageable);
```
Comment (FlxB2) [inline org.planqk.atlas.core/src/main/java/org/planqk/atlas/core/repository/AlgorithmRepository.java:48]: Please shorten the method name, as you can see [here](https://github.com/daggerok/spring-data-examples/blob/master/shadov/src/main/java/daggerok/ShadovApplication.java)

Comment (LHommeDeBat) [reply org.planqk.atlas.core/src/main/java/org/planqk/atlas/core/repository/AlgorithmRepository.java:52]: Cool, did not know you could do that.

**Pull request merged**